pkg/controller/pravega/pravega_controller.go

```diff
@@ -206,8 +208,53 @@ func makeControllerPodSpec(p *api.PravegaCluster) *corev1.PodSpec {
 	if p.Spec.Pravega.ControllerSecurityContext != nil {
 		podSpec.SecurityContext = p.Spec.Pravega.ControllerSecurityContext
 	}
+	podSpec.InitContainers = []corev1.Container{}
 	if p.Spec.Pravega.ControllerInitContainers != nil {
-		podSpec.InitContainers = p.Spec.Pravega.ControllerInitContainers
+		podSpec.InitContainers = append(podSpec.InitContainers, p.Spec.Pravega.ControllerInitContainers...)
+	}
+	if p.Spec.Pravega.AuthImplementations != nil {
+		authContainers := []corev1.Container{}
+		var srcPath, mountPath string
+		for i, plugin := range p.Spec.Pravega.AuthImplementations {
+			log.Printf("Anisha plugin is %v", plugin)
+			if plugin.PluginLocation != "" {
+				srcPath = plugin.PluginLocation
+			} else {
+				srcPath = "/plugins/implementation/*.jar"
+			}
+			if plugin.MountPath != "" {
+				mountPath = plugin.MountPath
+			} else {
+				mountPath = "/opt/pravega/pluginlib"
+			}
+			arg := "cp " + srcPath + " " + mountPath
+
+			authContainers = append(authContainers, corev1.Container{
+				Name:    "authplugin" + strconv.Itoa(i),
+				Image:   plugin.Image,
+				Command: []string{"/bin/sh"},
+				Args: []string{
+					"-cx",
+					arg,
+				},
+				VolumeMounts: []corev1.VolumeMount{
+					{
+						Name:      "authplugin" + strconv.Itoa(i),
```

Comment:
I believe the mountPath is static (and well known to the pravega controller), and there are, since we're in a loop, X volumeMounts getting mounted to the same mount path.    

Is that a problem (is one masking the other) or are they merged somehow (with /opt/pravega/pluginlib/  showing an aggregation of all the mounted "files").

Reply:
yes, it is an issue. mount path should be unique.   Modified the code to have unique mount path